Onboarding: draft implementation for tablet-landscape high-fi

MohamedMedhat1998 opened the pull request:
<!--
  - Thanks for submitting code to Oppia! Please fill out the following as part of
  - your pull request so we can review your code more easily.
  -->

## Explanation
This is a draft implementation for Onboarding tablet-landscape high-fi so that the mentor can take a look
<!--
  - Explain what your PR does. If this PR fixes an existing bug, please include
  - "Fixes #bugnum:" in the explanation so that GitHub can auto-close the issue
  - when this PR is merged.
  -->

## Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [ ] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [ ] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [ ] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [ ] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [ ] The PR is made from a branch that's **not** called "develop".
- [ ] The PR is made from a branch that is up-to-date with "develop".
- [ ] The PR's branch is based on "develop" and not on any other branch.
- [ ] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

MohamedMedhat1998 committed:
-Onboarding: draft implementation for tablet-landscape high-fi

rt4914 commented on app/src/main/res/drawable/ic_landscape_onboarding_4.xml:
```diff
@@ -0,0 +1,35 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Instead of this you can use the `ic_landscape_onboarding_4` as that is the more correct image.

MohamedMedhat1998 replied:
This is `ic_landscape_onboarding_4`, do you mean I should use `ic_landscape_onboarding_3` ?

MohamedMedhat1998 replied:
I have removed  `ic_landscape_onboarding_4` as it contains a problem and used `ic_landscape_onboarding_3` instead

rt4914 commented on app/src/main/res/layout-sw600dp-land/onboarding_slide.xml:
```diff
@@ -20,10 +20,10 @@
       android:src="@{viewModel.slideImage}"
       app:layout_constraintBottom_toBottomOf="parent"
-      app:layout_constraintDimensionRatio="H, 10:9"
+      app:layout_constraintEnd_toStartOf="@+id/guideline3"
```
Instead of this you should use `app:layout_constraintDimensionRatio `, that will solve your issue of the image.

MohamedMedhat1998 replied:
Ok, I'll give it a try

MohamedMedhat1998 replied:
Done!  
PTAL  
I made these changes:  
-Used `app:layout_constraintDimensionRatio="H, 10:9"` and `android:scaleType="fitXY"` instead of `guideline` and `android:scaleType="centerCrop"`
-Centered the `textViews` between the `imageView` and the end of the `parentLayout` and gave them a fixed width of `360dp` instead of placing the `textViews` relative to the end of the `parentLayout` only  
Please review them and tell me if there is something that needs to change

rt4914 replied:
We should avoid giving fixed widths. So mainly you will need to shift to relative positioning which is `96dp` as mentioned in the email.

Also, please convert this PR to final PR and then assign it back to me. Ideally a PR should be assigned as final PR for full review so that I can leave comments accordingly.

rt4914 reviewed (changes requested):
PTAL

MohamedMedhat1998 committed:
-Removed the unused drawable file
-Used `app:layout_constraintDimensionRatio="H, 10:9"` and `android:scaleType="fitXY"` instead of `guideline` and `android:scaleType="centerCrop"`
-Centered the `textViews` between the `imageView` and the end of the `parentLayout` and gave them a fixed width of `360dp` instead of placing the `textViews` relative to the end of the `parentLayout` only

MohamedMedhat1998 committed:
-placed the textViews and the buttons relative to the end of the screen (with a margin of 96dp) and a guideline of 57% from the beginning of the screen instead of using fixed width of 360dp

MohamedMedhat1998 commented:
Closed this PR because it is a draft PR.  
A new PR will be open instead of that
